iis-wf/iis-wf-report/src/main/java/eu/dnetlib/iis/wf/report/pushgateway/converter/CounterReportEntryMetricExtractor.java

```diff
@@ -0,0 +1,60 @@
+package eu.dnetlib.iis.wf.report.pushgateway.converter;
+
+import eu.dnetlib.iis.common.schemas.ReportEntry;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class CounterReportEntryMetricExtractor {
+
+    private CounterReportEntryMetricExtractor() {
+    }
+
+    @FunctionalInterface
+    public interface Extractor {
+        ExtractedMetric extract(ReportEntry reportEntry, Map<String, LabeledMetricConf> labeledMetricConfByPattern);
+    }
+
+    public static ExtractedMetric extractMetricFromReportEntry(ReportEntry reportEntry,
+                                                               Map<String, LabeledMetricConf> labeledMetricConfByPattern) {
+        return extractMetricFromReportEntry(reportEntry,
+                labeledMetricConfByPattern,
+                MetricNameCustomizer::dashRemover,
+                LabelValuesExtractor::extractLabelValuesByPosition);
+    }
+
+    public static ExtractedMetric extractMetricFromReportEntry(ReportEntry reportEntry,
+                                                               Map<String, LabeledMetricConf> labeledMetricConfByPattern,
+                                                               MetricNameCustomizer.Customizer metricNameCustomizer,
+                                                               LabelValuesExtractor.Extractor labelValuesExtractor) {
+        return labeledMetricConfByPattern.entrySet().stream()
+                .filter(entry -> keyMatchesPattern(reportEntry.getKey().toString(), entry.getKey()))
+                .findFirst()
```

Comment:
For labeled metric report entry's key should match only one regex from the map keys. Then the map will contain only one element and I've used `findFirst` to get that one entry. It could also be `findAny`. However, it is confusing so I've rewritten this code: first filer map entries matching report's key, then if there is an entry check if there's only one entry and extract metric using label metric conf; if there's no match return metric without labels.